### ISSUE TITLE
Non-public onionshare does not work

### ISSUE THREAD

**Issue by blabno** [onionshare/onionshare#1738]
When I run public onion service then I can access it through Tor Browser

```
onionshare-cli --public ~/OnionShare/
```

But if I run without `--public` then i'm constantly getting Unable to connect `ERR_SOCKS_CONNECTION_FAILED`

```
onionshare-cli ~/OnionShare/
```

**Comment by mig5** [comment]
Hi @blabno , there isn't quite enough info to go on here. You could add `--verbose` to obtain maybe a bit more debug info as to what might be happening - just be careful if you paste any of that info here, since it might contain sensitive info to do with your onion URL or private key.

The error `ERR_SOCKS_CONNECTION_FAILED` suggests potentially an issue with the version of Tor you are using on your system. It might be too old to support Client Authentication (private keys). Please share some more info about `tor --version` and the OS/version of the OS.

**Comment by blabno** [comment]
I'm using Tor version 0.4.7.13.

```
onionshare-cli --verbose ~/OnionShare/
OnionShare v2.6
https://onionshare.org/

[Sep 07 2023 09:36:06] Settings.__init__
[Sep 07 2023 09:36:06] Settings.load
[Sep 07 2023 09:36:06] Common.get_resource_path: filename=wordlist.txt, path=/home/user/.local/lib/python3.10/site-packages/onionshare_cli/resources/wordlist.txt
[Sep 07 2023 09:36:06] ModeSettings.load: creating /home/user/.config/onionshare/persistent/smokeless-shield-curing.json
[Sep 07 2023 09:36:06] ModeSettings.set: updating smokeless-shield-curing: general.title = None
[Sep 07 2023 09:36:06] ModeSettings.set: updating smokeless-shield-curing: general.public = False
[Sep 07 2023 09:36:06] ModeSettings.set: updating smokeless-shield-curing: general.autostart_timer = 0
[Sep 07 2023 09:36:06] ModeSettings.set: updating smokeless-shield-curing: general.autostop_timer = 0
[Sep 07 2023 09:36:06] ModeSettings.set: updating smokeless-shield-curing: share.autostop_sharing = True
[Sep 07 2023 09:36:06] Web.__init__: is_gui=False, mode=share
[Sep 07 2023 09:36:06] Common.get_resource_path: filename=static, path=/home/user/.local/lib/python3.10/site-packages/onionshare_cli/resources/static
[Sep 07 2023 09:36:06] Common.get_resource_path: filename=templates, path=/home/user/.local/lib/python3.10/site-packages/onionshare_cli/resources/templates
[Sep 07 2023 09:36:06] Web.generate_static_url_path: new static_url_path is /static_7imnorgg7s7qwwftns4yvnxylu
[Sep 07 2023 09:36:06] ShareModeWeb.init
[Sep 07 2023 09:36:06] Onion.__init__
[Sep 07 2023 09:36:06] Settings.__init__
[Sep 07 2023 09:36:06] Settings.load
[Sep 07 2023 09:36:06] Onion.connect: connection_type=bundled
[Sep 07 2023 09:36:06] Onion.connect: tor_data_directory_name=/home/user/.config/onionshare/tmp/tmpccnjy45b
[Sep 07 2023 09:36:06] Common.get_resource_path: filename=torrc_template, path=/home/user/.local/lib/python3.10/site-packages/onionshare_cli/resources/torrc_template
[Sep 07 2023 09:36:06] Onion.connect: Writing torrc template file
[Sep 07 2023 09:36:06] Onion.connect: starting /usr/bin/tor subprocess
[Sep 07 2023 09:36:06] Onion.connect: tor pid: 8620
[Sep 07 2023 09:36:08] Onion.connect: authenticating to tor controller
Connecting to the Tor network: 100% - Done
[Sep 07 2023 09:36:34] Onion.connect: Connected to tor 0.4.7.13
[Sep 07 2023 09:36:34] Settings.load
[Sep 07 2023 09:36:34] OnionShare.__init__
[Sep 07 2023 09:36:34] OnionShare.start_onion_service
[Sep 07 2023 09:36:34] Onion.start_onion_service: port=17640
[Sep 07 2023 09:36:34] Onion.start_onion_service: key_type=NEW, key_content=ED25519-V3
[Sep 07 2023 09:36:37] ModeSettings.set: updating smokeless-shield-curing: general.service_id = 
[Sep 07 2023 09:36:37] ModeSettings.set: updating smokeless-shield-curing: onion.private_key = 
[Sep 07 2023 09:36:37] ModeSettings.set: updating smokeless-shield-curing: onion.client_auth_priv_key = 
[Sep 07 2023 09:36:37] ModeSettings.set: updating smokeless-shield-curing: onion.client_auth_pub_key = 
Compressing files.
[Sep 07 2023 09:36:37] ShareModeWeb.init
[Sep 07 2023 09:36:37] ShareModeWeb.set_file_info_custom
[Sep 07 2023 09:36:37] ShareModeWeb.build_zipfile_list: filenames=['/home/user/OnionShare/test2.md', '/home/user/OnionShare/test.md', '/home/user/OnionShare/2023-08-23']
[Sep 07 2023 09:36:37] Web.start: port=17640
 * Running on http://127.0.0.1:17640/ (Press CTRL+C to quit)

 Give this address and private key to the recipient:
 http://7gecyygtsmgws7yrd4kdp27ggay3z4glc3dd6xrntyavn4r3ux7jp5id.onion
 Private key: COS2JKACVLRAQKLCDFIXJ2ACXAD6Q74WLYPPZAIKRUTUF2TYRMUA

 Press Ctrl+C to stop the server
 ```
 
I've cut out
* general.service_id 
* onion.private_key 
* onion.client_auth_priv_key 
* onion.client_auth_pub_key

**Comment by mig5** [comment]
@blabno I can't see any errors with OnionShare starting the service. (note the private key/onion were still shown at the end of the output, I've masked that for you but recommend you don't use that .onion or private key long term if sensitive).

So is it that you're saying that you get `ERR_SOCKS_CONNECTION_FAILED` in *Tor Browser* when you go to view the service?

If so, what's the Tor Browser version, and the OS?

Do you reproduce it with Tor Browser on other devices, if you have access to any and can install it there?

In summary, it sounds like it might just be a Tor connection issue in your browser rather than an OnionShare problem.

If you are using Whonix, note that its design between 'workstation' and 'gateway' means you need to add the private key in a different way than via the browser UI (as you would in native Tor Browser on a non-Whonix OS) - see one of the two options shown at https://www.whonix.org/wiki/Tor_Browser#Onion_Client_Authorization

**Comment by blabno** [comment]
I'm testing with
* Brave Version 1.47.164 Chromium: 109.0.5414.74 (Official Build) running on same device as OnionShare
* Tor browser (on Android) 102.2.1-Release (12.5.3)  running on mobile device

Again, this cannot be about connection, because I am able to connect to OnionShare if I use `--public` flag.

**Comment by mig5** [comment]
Yeah, so my suspicion is that those browsers don't have a way to support 'client auth' (whereby you enter the 'private key' that OnionShare gives you)

That's certainly true for the Tor Browser on Android, apparently: https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/31672

I can't find info on Brave, but I suspect it's the same issue.

Using the official Tor Browser on a desktop would work with Client Auth - you'd get a prompt to enter the private key and then you'd be let in.

The reason it works for you with `--public` is because that explicitly disables Client Auth (that's why there's no 'private key' given to you when you use that mode).

Everything in your OnionShare is working fine - it's just the client apps you're using to access Tor, don't seem to support Tor's Client Auth feature.

Hope that helps!

**Comment by blabno** [comment]
Using standard desktop TorBrowser I can access non-publick OnionShare. Thank you @mig5.
